TESTING.md

```diff
@@ -0,0 +1,114 @@
+leave test data afterwards.** Most tests create data in a
+well-partitioned way and clean up after themselves, but there could
+be a bug in a test which wipes all the data in the project. It is
+*very strongly* recommended that you only run tests against projects
+which do not contain any production data or services.
+
+Running tests
+=============
+
+The `build.sh` script will run unit tests, and
+`runintegrationtests.sh` will run integration tests and snippets.
+(TODO: smoke tests too...) Consult the scripts for details on how to
+run.
+
+Additionally, you can just use `dotnet test` in any test project,
+optionally specifying a test filter and/or target framework.
+
+Common environment variables
+============================
+
+To run the integration/smoke tests or snippets, set the following
+environment variables:
+
+- `TEST_PROJECT`: The name of the Google Cloud Project to test against
+- `GOOGLE_APPLICATION_CREDENTIALS`: The absolute path to a JSON file
+  containing service credentials for the test project.
+
+Per API configuration
+=====================
+
+Some APIs require a little more setup. If you're not testing an API
+listed below, the instructions above should be all you need.
+
+Google.Cloud.Bigtable.V2
+------------------------
+
+The Bigtable integration tests can be run against the production
+service or the emulator.
+
+Running in production requires:
+
+- A Bigtable instance, preferably of type Development, created within the
+  project and identified by the `BIGTABLE_TEST_INSTANCE` environment variable
+- The account identified in your service account JSON file to have read/write
+  access to the project, including the IAM permissions in the Bigtable Administrator role
+
+Running against the emulator requires:
+
+- A running emulator on the local machine, started by using the
+  `gcloud beta emulators bigtable start` command
+- The `BIGTABLE_EMULATOR_HOST` environment variable set to `localhost:8086`
+
+Currently there are no runnable snippets, but when they are added, they will
+have the same requirements.
+
+Google.Cloud.Firestore
+----------------------
+
+Firestore cannot currently operate on a Google Cloud project that also has a
+Datastore database. You will need a second project:
+
+- Create a project on the [Firebase Console](https://console.firebase.google.com/u/0/)
+- Enable Firestore on that project (within the Firebase Console)
+- Set the `FIRESTORE_TEST_PROJECT` environment variable with that
+  project ID
+- Add the account identified in your service account JSON file as an
+  owner of the project
+  
+Google.Cloud.Storage.V1
+-----------------------
+
+- Enable the Google Cloud Pub/Sub API as well as Google Cloud Storage
+- Determine your Google Cloud Storage service account and give it
+  publisher access for PubSub:
+  - Either run the integration tests and see the error message in
+    the failure, or determine the service account as
+    `PROJECT_ID_HERE@gs-project-accounts.iam.gserviceaccount.com`
```

Comment:
Is this project id or project number?

Reply:
The project ID, same as `TEST_PROJECT`. Would "where `PROJECT_ID_HERE` is the value of the `TEST_PROJECT` environment variable" help?

Reply:
I don't think it matters too much.  I just know most service accounts use the project number instead of the project id.  But if this just uses the ID what you have is good.